### ISSUE TITLE
Flatten exports to ONNX in a non-portable manner

### ISSUE THREAD

**Issue by jlucier** [qubvel/segmentation_models.pytorch#485]
Firstly, thank you for the great library! This is an awesome resource!

----

## Description
Anyhow, when using the provided mechanism to add a `ClassificationHead` to our networks, we noticed that the head exports ONNX in a bit of an unexpected way (screenshot of ONNX layers below). The reason is because of this [code](https://github.com/qubvel/segmentation_models.pytorch/blob/master/segmentation_models_pytorch/base/modules.py#L123L125) used for `Flatten` in the `ClassificationHead`:
```python
class Flatten(nn.Module):
    def forward(self, x):
        return x.view(x.shape[0], -1)
```

This isn't immediately a problem because the resulting ONNX model is after all a valid one. However, we use TensorRT as an inference runtime and TensorRT very much does not like this configuration (specifically I think the `Cast` / `Constant` things happening here).

![Screen Shot 2021-09-14 at 10 51 06 AM](https://user-images.githubusercontent.com/5672903/133282417-368f3cb8-fd1a-471f-9943-6df679b46ba5.png)

## Proposed Solution
The functionality provided by the custom `Flatten` you have defined in this library seems to be:
> Flatten all dimensions of the input besides the first (presumably the batch) dimension.

This is actually identical to how the built-in [`torch.nn.Flatten`](https://pytorch.org/docs/stable/generated/torch.nn.Flatten.html?highlight=flatten#torch.nn.Flatten) works. If you remove the custom `Flatten` layer and instead use `torch`'s, the exported ONNX will have an ONNX `Flatten` operation in it, which is both more portable to other runtimes and more closely matches the intended behavior / structure (depicted below -- same network as before, changed your `Flatten` to `torch.nn.Flatten`).

I could have missed a reason why you would prefer your own implementation of `Flatten`, but assuming there isn't a particularly pressing one I think it would be a good move to migrate to the built-in `torch.nn.Flatten`. Happy to make that PR, but figured I would bring it to light first before writing code.

![Screen Shot 2021-09-14 at 11 27 41 AM](https://user-images.githubusercontent.com/5672903/133287147-6b66b45f-7d0e-487c-8b15-0c440c940b83.png)

**Comment by qubvel** [comment]
Thanks for reporting.
I have updated master branch, please, check

**Comment by jlucier** [comment]
Looks good! Thanks @qubvel

**Comment by jlucier** [comment]
Solved with #486
